seismostats/analysis/estimate_mc.py

```diff
@@ -69,17 +71,41 @@ def empirical_cdf(
     except BaseException:
         pass
 
+    if get_option("warnings") is True:
+        # check if binning is correct
+        if not np.allclose(sample, bin_to_precision(sample, delta_m)):
+            warnings.warn(
+                "Magnitudes are not binned correctly. "
+                "Test might fail because of this."
+            )
+
+    if mc is None:
+        mc = np.min(sample)
+
     idx = np.argsort(sample)
-    sample_sorted = sample[idx]
+    x = sample[idx]
 
     if weights is not None:
         weights_sorted = weights[idx]
-        x, y = sample_sorted, np.cumsum(weights_sorted) / weights_sorted.sum()
+        y = np.cumsum(weights_sorted) / weights_sorted.sum()
     else:
-        x, y = sample_sorted, np.arange(1, len(sample) + 1) / len(sample)
+        y = np.arange(1, len(sample) + 1) / len(sample)
 
     x, y_count = np.unique(x, return_counts=True)
+    # add empty bins
+    if delta_m > 0:
+        for mag_bin in bin_to_precision(
+            np.arange(mc, np.max(sample) + delta_m, delta_m), delta_m
+        ):
+            if mag_bin not in x:
+                x = np.append(x, mag_bin)
+                y_count = np.append(y_count, 0)
+        idx = np.argsort(x)
+        x = x[idx]
+        y_count = y_count[idx]
+
     y = y[np.cumsum(y_count) - 1]
```

Comment:
np.cumsum(y_count) can be zero at the beginning.
Then, we want the corresponding y to be 0